### ISSUE TITLE
Unable to find vtable symbol for virtual function <std::alloc::LayoutError as std::fmt::Debug>::fmt

### ISSUE THREAD

**Issue by YoshikiTakashima** [model-checking/kani#1321]
I tried this code:
[tests/kani/NondetSlices/test2.rs](https://github.com/model-checking/kani/blob/92e5889d2d81424cc396c61df4c5480e935ea2c4/tests/kani/NondetSlices/test2.rs)
using the following command line invocation:

```
kani tests/kani/NondetSlices/test2.rs
```

with Kani version: 0.5.0 on M1 Mac

I expected to see **no errors**
Instead, this happened: warnings about vtable 
```
 WARN kani_compiler::codegen_cprover_gotoc::codegen::rvalue Unable to find vtable symbol for virtual function <std::alloc::LayoutError as std::fmt::Debug>::fmt, attempted lookup for symbol name: _ZN69_$LT$core..alloc..layout..LayoutError$u20$as$u20$core..fmt..Debug$GT$3fmt17h6ac16430b0dcd371E
```
